.github/workflows/haskell.yml

```diff
@@ -0,0 +1,126 @@
+name: Haskell CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
```

Comment:
Not for this PR, but I wonder if we want to invest time in abstracting away this CI setup we replicate in multiple repos.

Reply:
I will create an issue for this